pkg/operator/controller/externaldns/status_test.go

```diff
@@ -0,0 +1,634 @@
+			cond := computeDeploymentPodsScheduledCondition(context.TODO(), cl, &tc.existingDeployment)
+			if diff := cmp.Diff(tc.expectedResult, cond, ignoreTimeOpt); diff != "" {
+				t.Errorf("expected condition %v; got condition %v: \n %s", tc.expectedResult, cond, diff)
+			}
+		})
+	}
+}
+
+func TestMergeConditions(t *testing.T) {
+	testCases := []struct {
+		name               string
+		existingConditions []metav1.Condition
+		updateCondition    metav1.Condition
+		expectedResult     []metav1.Condition
+	}{
+		{
+			name:               "Starting by empty conditions, with 1 update, results in 1 condition",
+			existingConditions: []metav1.Condition{},
+			updateCondition: metav1.Condition{
+				Type:    ExternalDNSDeploymentAvailableConditionType,
+				Status:  metav1.ConditionUnknown,
+				Reason:  "DeploymentAvailabilityUnknown",
+				Message: "The deployment has no Available status condition set",
+			},
+			expectedResult: []metav1.Condition{
+				{
+					Type:    ExternalDNSDeploymentAvailableConditionType,
+					Status:  metav1.ConditionUnknown,
+					Reason:  "DeploymentAvailabilityUnknown",
+					Message: "The deployment has no Available status condition set",
+				},
+			},
+		},
+		{
+			name: "Starting with 1 condition, with 1 updated condition, results in 1 condition",
+			existingConditions: []metav1.Condition{
+				{
+					Type:    ExternalDNSDeploymentAvailableConditionType,
+					Status:  metav1.ConditionTrue,
+					Reason:  "DeploymentAvailable",
+					Message: "The deployment has Available status condition set to True",
+				},
+			},
+			updateCondition: metav1.Condition{
+				Type:    ExternalDNSDeploymentAvailableConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  "DeploymentUnavailable",
+				Message: "The deployment has Available status condition set to False",
+			},
+			expectedResult: []metav1.Condition{
+				{
+					Type:    ExternalDNSDeploymentAvailableConditionType,
+					Status:  metav1.ConditionFalse,
+					Reason:  "DeploymentUnavailable",
+					Message: "The deployment has Available status condition set to False",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			conditions := mergeConditions(tc.existingConditions, tc.updateCondition)
+			if len(conditions) != len(tc.expectedResult) {
+				t.Errorf("expected number of conditions %v; got %v conditions", len(tc.expectedResult), len(conditions))
+			}
+			for _, expectedCondition := range tc.expectedResult {
+				isConditionTypeFound := false
+				for _, cond := range conditions {
+					if cond.Type == expectedCondition.Type {
+						isConditionTypeFound = true
+						if diff := cmp.Diff(expectedCondition, cond, ignoreTimeOpt); diff != "" {
+							t.Errorf("expected condition %v; got condition %v: \n %s", tc.expectedResult, cond, diff)
+						}
+					}
+				}
+				if !isConditionTypeFound {
+					t.Errorf("expected condition type %v was not found in the result", expectedCondition.Type)
+				}
+			}
```

Comment:
All this can be replaced by a single Diff:
```go
	conditionCmpOpts := []cmp.Option{
		cmpopts.EquateEmpty(),
		ignoreTimeOpt,
		cmpopts.SortSlices(func(a, b metav1.Condition) bool { return a.Type < b.Type }),
	}
	if diff := cmp.Diff(expectedCondition, cond, conditionCmpOpts...); diff != "" {
	}
```